ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/ReplaceWithReferenceExprFixBase.fs

```diff
@@ -0,0 +1,32 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Resources.Shell
+
+[<AbstractClass>]
+type ReplaceWithReferenceExprFixBase(expr: IFSharpExpression, refExprName, refExprFullName) =
+    inherit FSharpQuickFixBase()
+    
+    abstract member ResolveContext: ITreeNode
+    abstract member BeforeReplaceExecute: unit -> unit
+    default x.BeforeReplaceExecute() = ()
+    
+    member x.ExprToReplace = expr
+
+    override x.IsAvailable _ =
+        isValid expr &&
+        match expr.CheckerService.ResolveNameAtLocation(x.ResolveContext, [refExprName], "ReplaceWithReferenceExprFixBase") with
+        | Some symbolUse -> symbolUse.Symbol.FullName = refExprFullName
```

Comment:
What will happen if there's no `FullName` for a symbol?
Consider local values, for instance:
```fsharp
let f id =
  fun x -> x
```

Reply:
@auduchinok, this will work since for the specified case FullName = "id"